Changes for TrkAna V5

brownd1978 opened the pull request:
Replace some includes with forward declaration to avoid exposing TrkAna to art.  Update some test scripts.

FNALbuild commented:
Hi @brownd1978,
You have proposed changes to files in these packages:
- Mu2eKinKal
- MCDataProducts

which require these tests: build.

@Mu2e/fnalbuild-users, @Mu2e/write have access to CI actions on [main](/Mu2e/Offline/tree/main).





:hourglass: The following tests have been triggered for 4471488b764ed8ab43d252cac09fc3cec89a35f4: build  (Build queue is empty)


[About FNALbuild.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#GitHub_Pull_Request_Procedures_and_FNALbuild) [Code review on Mu2e/Offline.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#Code_Review)

FNALbuild commented:
:umbrella: The build is failing at 4471488b764ed8ab43d252cac09fc3cec89a35f4.

```
scons: *** [build/sl7-prof-e28-p055/Offline/tmp/MCDataProducts/src/MCRelationship.os] Error 1
```


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged 4471488b764ed8ab43d252cac09fc3cec89a35f4 at c5b77ada16857f31eebf84bc8f6001e18089d84f |
| build (prof) | :x: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/scons.log).  |
| ceSimReco | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/ceSimReco.log)  |
| g4test_03MT | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/g4test_03MT.log)  |
| transportOnly | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/transportOnly.log)  |
| POT | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/POT.log)  |
| g4study | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/g4study.log)  |
| cosmicSimReco | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/cosmicOffSpill.log)  |
| ceSteps | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/ceSteps.log)  |
| ceDigi | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/ceDigi.log)  |
| muDauSteps | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/muDauSteps.log)  |
| ceMix | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/ceMix.log)  |
| rootOverlaps | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :wavy_dash: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [3 errors 29 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at 4471488b764ed8ab43d252cac09fc3cec89a35f4 after being merged into the base branch at c5b77ada16857f31eebf84bc8f6001e18089d84f.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2040/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

brownd1978 commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for 4bf7bbf1c2dc91d27f1561e51651b05cc18b2870: build  (Build queue is empty)

FNALbuild commented:
:sunny: The build tests passed at 4bf7bbf1c2dc91d27f1561e51651b05cc18b2870.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged 4bf7bbf1c2dc91d27f1561e51651b05cc18b2870 at c5b77ada16857f31eebf84bc8f6001e18089d84f |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/scons.log). Build time: 10 min 58 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 29 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at 4bf7bbf1c2dc91d27f1561e51651b05cc18b2870 after being merged into the base branch at c5b77ada16857f31eebf84bc8f6001e18089d84f.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2041/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
@rlcee I gentle reminder that his PR is on your list to review.

kutschke commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for d3cfd71b3867d960c8b16c22414f4670eccc6a4d: build  (Build queue is empty)

brownd1978 commented:
I am addressing the comment made by rlc and will have an update soon, so
please hold off merging this PR

On Wed, Mar 27, 2024 at 8:02 AM Fermilab build bot account <
***@***.***> wrote:

> ⌛ The following tests have been triggered for d3cfd71
> <https://github.com/Mu2e/Offline/commit/d3cfd71b3867d960c8b16c22414f4670eccc6a4d>:
> build (Build queue is empty)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/1225#issuecomment-2023001292>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH576M4ECPJGEDBV6KZDDY2LNRJAVCNFSM6AAAAABFH565IWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMRTGAYDCMRZGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
David Brown            ***@***.***
Office Phone (510) 486-7261
Lawrence Berkeley National Lab
M/S 50R5008 (50-6026C)  Berkeley, CA 94720

FNALbuild commented:
:sunny: The build tests passed at d3cfd71b3867d960c8b16c22414f4670eccc6a4d.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged d3cfd71b3867d960c8b16c22414f4670eccc6a4d at c5b77ada16857f31eebf84bc8f6001e18089d84f |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/scons.log). Build time: 11 min 35 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 29 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at d3cfd71b3867d960c8b16c22414f4670eccc6a4d after being merged into the base branch at c5b77ada16857f31eebf84bc8f6001e18089d84f.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2043/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
> I am addressing the comment made by rlc and will have an update soon, so please hold off merging this PR

Is this ready to go now?

brownd1978 commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for fe300a8f3fd3ec748d7f83f58338c9b2217e0a44: build  (Build queue is empty)

FNALbuild commented:
:sunny: The build tests passed at fe300a8f3fd3ec748d7f83f58338c9b2217e0a44.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged fe300a8f3fd3ec748d7f83f58338c9b2217e0a44 at c5b77ada16857f31eebf84bc8f6001e18089d84f |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/scons.log). Build time: 10 min 59 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 29 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at fe300a8f3fd3ec748d7f83f58338c9b2217e0a44 after being merged into the base branch at c5b77ada16857f31eebf84bc8f6001e18089d84f.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2044/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
I looked at the clang-tidy report.  Most of the errors fellow travellers that are not part of this PR.  And they are mostly in code that I wrote.

The one item that is in the files touched by this PR is the 4 messages from Offline/MCDataProducts/src/MCRelationship.cc  that relate to setting _rem.  Should we make a plan to address those in a followup PR?

I will clean up the errors that come from GenId, PDGCode and SimParticle in a separate PR.

brownd1978 commented:
I looked at the _rem code suggestions, they might be improvements but it is
very risky to change that logic without extensive testing (it took a few
iterations to get it right).  I don't see a big win and suggest to ignore
them.

On Thu, Mar 28, 2024 at 10:39 AM Rob Kutschke ***@***.***>
wrote:

> I looked at the clang-tidy report. Most of the errors fellow travellers
> that are not part of this PR. And they are mostly in code that I wrote.
>
> The one item that is in the files touched by this PR is the 4 messages
> from Offline/MCDataProducts/src/MCRelationship.cc that relate to setting
> _rem. Should we make a plan to address those in a followup PR?
>
> I will clean up the errors that come from GenId, PDGCode and SimParticle
> in a separate PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/1225#issuecomment-2025770947>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH5727Y4CIAPUYF7D5FDDY2RITJAVCNFSM6AAAAABFH565IWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMRVG43TAOJUG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
David Brown            ***@***.***
Office Phone (510) 486-7261
Lawrence Berkeley National Lab
M/S 50R5008 (50-6026C)  Berkeley, CA 94720

kutschke commented:
OK - we can decide to address them later or not.  I will merge now.
